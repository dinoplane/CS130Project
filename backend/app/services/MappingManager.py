from backend.app.schema.mapping_schema import MappingEntry
from fastapi.encoders import jsonable_encoder as jsn_enc
from fastapi import Request, Response, Body, status
from backend.app.database.mongodb import insert_one, find_one, find

class MappingDB_Manager():
    def add_mapping(req: Request, mapping = MappingEntry):
        mapping = jsn_enc(mapping)
        #insert to DB
        new_map = insert_one(req, mapping)
         #return inserted val from DB (i.e. ensure it got added)
        map_add = find_one(req, new_map)
        return map_add
    def fetch_mappings(req: Request):
        #something to get the mappings
        mappingEntries = find(req)
        return mappingEntries
    def remove_mapping():
        return
        #something to remove mapping
    def validate_mapping():
        return
        #something to validate mapping
