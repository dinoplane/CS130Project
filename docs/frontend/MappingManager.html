<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>JSDoc: Class: MappingManager</title>

    <script src="scripts/prettify/prettify.js"></script>
    <script src="scripts/prettify/lang-css.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link
      type="text/css"
      rel="stylesheet"
      href="styles/prettify-tomorrow.css"
    />
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css" />
  </head>

  <body>
    <div id="main">
      <h1 class="page-title">Class: MappingManager</h1>

      <section>
        <header>
          <h2>
            <span class="attribs"><span class="type-signature"></span></span
            >MappingManager<span class="signature">()</span
            ><span class="type-signature"></span>
          </h2>
        </header>

        <article>
          <div class="container-overview">
            <h4 class="name" id="MappingManager">
              <span class="type-signature"></span>new MappingManager<span
                class="signature"
                >()</span
              ><span class="type-signature"></span>
            </h4>

            <dl class="details">
              <dt class="tag-source">Source:</dt>
              <dd class="tag-source">
                <ul class="dummy">
                  <li>
                    <a href="mapping_manager.js.html">mapping_manager.js</a>,
                    <a href="mapping_manager.js.html#line1">line 1</a>
                  </li>
                </ul>
              </dd>
            </dl>
          </div>
        </article>
      </section>
    </div>

    <nav>
      <h2><a href="index.html">Home</a></h2>
      <h3>Modules</h3>
      <ul>
        <li><a href="module-DropdownMenu.html">DropdownMenu</a></li>
      </ul>
      <h3>Classes</h3>
      <ul>
        <li><a href="ExcelHandler.html">ExcelHandler</a></li>
        <li><a href="MappingManager.html">MappingManager</a></li>
        <li><a href="module.exports.html">exports</a></li>
        <li><a href="module.exports_module.exports.html">exports</a></li>
      </ul>
      <h3>Global</h3>
      <ul>
        <li><a href="global.html#downloadExcel">downloadExcel</a></li>
        <li><a href="global.html#setFusekiUrl">setFusekiUrl</a></li>
        <li><a href="global.html#uploadExcel">uploadExcel</a></li>
      </ul>
    </nav>

    <br class="clear" />

    <footer>
      Documentation generated by
      <a href="https://github.com/jsdoc/jsdoc">JSDoc 4.0.2</a> on Fri Dec 08
      2023 00:23:40 GMT-0800 (Pacific Standard Time)
    </footer>

    <script>
      prettyPrint();
    </script>
    <script src="scripts/linenumber.js"></script>
  </body>
</html>
