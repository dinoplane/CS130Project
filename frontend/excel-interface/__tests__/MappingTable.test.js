import {fireEvent, render, screen} from "@testing-library/react"

// import {render} from '@testing-library/react'
import MappingTable from "@/app/mapping_table";
import MappingManager from "@/app/mapping_manager";


const mappingManager = new MappingManager();
const MAPPINGS = [
    {
      id: 0,
      mapping_query: "SELECT STUDENT WHERE COURSE.NAME = CS130",
      date_modified: "10/19/2023",
    },
    {
      id: 1,
      mapping_query: "SELECT COURSE WHERE STUDENTS_ENROLLED > 30 AND DEPT = CS",
      date_modified: "10/20/2023",
    },
    {
      id: 2,
      mapping_query: "SELECT PROFESSOR WHERE COURSE.NAME = CS130",
      date_modified: "10/21/2023",
    },
  ];

it('adding a row', () => {
    const component = render(
        <MappingTable mappings={MAPPINGS} mappingManager={mappingManager} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // console.log(component)
    let addrow_button = component.container.querySelector('#addBtn');
    // console.log(addrow_button)

    fireEvent.click(addrow_button);

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();


    // console.log(tree.props)

    // // Open a template
    // renderer.act(
    //     () => {

    //     }
    // )

    // // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();


    //     // manually trigger the callback
    // renderer.act(() => {
    //     tree.props.addRow("SELECT PROFESSOR WHERE COURSE.NAME = CS130");
    // });

    // // re-rendering
    // tree = component.toJSON();
    // expect(tree).toMatchSnapshot();
});


