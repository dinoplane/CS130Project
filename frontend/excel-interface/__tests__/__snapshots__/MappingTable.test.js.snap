// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adding a row 1`] = `
<main
  className="main"
>
  <div
    className="title_bar"
  >
    <h1>
      Excellent Interface
    </h1>
    <button
      className="connect_button"
    >
      Connected to: URL
    </button>
  </div>
  <div
    className="datatable"
  >
    <div
      className="toolbar"
    >
      <div
        className="toolbar_button"
        onClick={[Function]}
      >
        <img
          alt="HAI"
          className="image"
          stroke="#333333"
        />
      </div>
      <div
        className="toolbar_button"
        onClick={[Function]}
      >
        <label
          htmlFor="file-input"
        >
          <img
            alt="HAI"
            className="image"
            stroke="#333333"
          />
        </label>
        <input
          id="file-input"
          onChange={[Function]}
          type="file"
        />
      </div>
      <div
        className="toolbar_button"
        onClick={[Function]}
      >
        <img
          alt="HAI"
          className="image"
          stroke="#df2323"
        />
      </div>
    </div>
    <table>
      <thead>
        <tr>
          <th>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
          </th>
          <th>
            ID
          </th>
          <th>
            Mapping Query
          </th>
          <th>
            Date Modified
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="norow"
        >
          <td
            className="checkbox_col"
          />
          <td
            className="id_col"
          />
          <td
            className="query_col"
          >
            Nothing to show here!
          </td>
          <td
            className="date_col"
          />
        </tr>
        <tr
          className="addrow"
        >
          <td />
          <td
            className="addrow_td"
          >
            <div
              className="toolbar_button"
              id="addBtn"
              onClick={[Function]}
            >
              <img
                className="image"
                stroke="#333333"
              />
            </div>
          </td>
          <td />
          <td />
        </tr>
      </tbody>
    </table>
  </div>
</main>
`;

exports[`adding a row 2`] = `
<div
  className="datatable"
>
  <div
    className="toolbar"
  >
    <div
      className="toolbar_button"
      onClick={[Function]}
    >
      <img
        alt="HAI"
        className="image"
        stroke="#333333"
      />
    </div>
    <div
      className="toolbar_button"
      onClick={[Function]}
    >
      <label
        htmlFor="file-input"
      >
        <img
          alt="HAI"
          className="image"
          stroke="#333333"
        />
      </label>
      <input
        id="file-input"
        onChange={[Function]}
        type="file"
      />
    </div>
    <div
      className="toolbar_button"
      onClick={[Function]}
    >
      <img
        alt="HAI"
        className="image"
        stroke="#df2323"
      />
    </div>
  </div>
  <table>
    <thead>
      <tr>
        <th>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
        </th>
        <th>
          ID
        </th>
        <th>
          Mapping Query
        </th>
        <th>
          Date Modified
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="norow"
      >
        <td
          className="checkbox_col"
        />
        <td
          className="id_col"
        />
        <td
          className="query_col"
        >
          Nothing to show here!
        </td>
        <td
          className="date_col"
        />
      </tr>
      <tr
        className="addrow"
      >
        <td />
        <td
          className="addrow_td"
        >
          <div
            className="toolbar_button"
            id="addBtn"
            onClick={[Function]}
          >
            <img
              className="image"
              stroke="#333333"
            />
          </div>
        </td>
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
